openml/datasets/functions.py

```diff
@@ -418,22 +416,62 @@ def create_dataset(name, description, creator, contributor, collection_date,
         'data': data
     }
 
+    # Determine arff format from the dataset
+    if isinstance(data, list):
+        if isinstance(data[0], list):
```

Comment:
Actually you need to check all the element of data. If I pass:

```python
data = [[1, 2], {'whatever': [0, 1]}]
```
this will not raise an error.

Reply:
Theoretically the liac-arff should throw an error regarding this. Just to be sure, I will check later "today". If it does, then I would probably try to avoid going through each row of the data.
The code above just wants to infer the format, I added the error raising statements since that is information that we already have from deciding on the format.

Reply:
OK so now it makes sense to let the construction of the `arff_object` above since it will serve of checking for the data variable.

Reply:
I checked and liac-arff does throw an error for the example above. I would say no, it is worth it to keep the checks which determine the format of the ARFF before the object creation, that way if the most simplest of cases are violated we throw an error and there is no arff object to be collected from gc.
Liac-arff can take care of the rest.